app/models/past_link.rb

```diff
@@ -14,7 +15,48 @@ def self.log_link(link)
         })
   end
 
+
+  def set_reaction (response_type, response_text)
```

Comment:
this is probably the most interesting thing that's actually reviewable here. this logic has been moved to the past_link model